action/protocol/execution/evm/evmstatedbadapter_test.go

```diff
@@ -377,9 +377,13 @@ func TestSnapshotRevertAndCommit(t *testing.T) {
 		//[TODO] need e2etest to verify state factory commit/re-open (whether result from state/balance/suicide/exist is same)
 	}
 
-	t.Run("contract snapshot/revert/commit with in memery DB", func(t *testing.T) {
+	t.Run("contract snapshot/revert/commit with in memory DB", func(t *testing.T) {
 		cfg := config.Default
-		testSnapshotAndRevert(cfg, t)
+		testSnapshotAndRevert(cfg, t, false)
+	})
+	t.Run("contract snapshot/revert/commit with async trie in memory DB", func(t *testing.T) {
+		cfg := config.Default
+		testSnapshotAndRevert(cfg, t, true)
```

Comment:
nit: is it possible to verify with `async = true, fixSnapshotOrder = false`, it will lead to that "failed to load root" error?

if not easy to modify in `testSnapshotAndRevert`, maybe add a new small test